### ISSUE TITLE
getAnimationStyle causes exceptions in older browsers

### ISSUE THREAD

**Issue by janousek** [angular/angular#24094]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

Errors are thrown during collection of animation styles (in older browsers).
The problem is on this line:
https://github.com/angular/angular/blob/13cb75da8b02ed7aa60508dbb5ecd9065809fd12/packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts#L137
Most of browsers will return empty string if the style property is not set. But some browsers (like Chrome on Android 4.4) returns null instead of empty string.
Several string functions are applied on the output of the method "getAnimationStyle" - like trim, split, indexOf, etc... But if the output of the"getAnimationStyle"  is null, exception is thrown.

Simple fix of this bug:
```
  return element.style[ANIMATION_PROP + name] || '';
```

## Expected behavior

Animation should work without errors.

## Environment

<pre><code>
Angular version: 6.0.3

Browser:
- [ ] Chrome (desktop) version XX
- [X] Chrome (Android) version 34
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by omieliekh** [comment]
Any progress on this issue? It now causes issues with PhantomJS tests

**Comment by janousek** [comment]
It seems that no one cares about this issue. My temporary solution/workaround to prevent errors looks like this:
```
@Component({
  ....
})
class AppComponent {

	@HostBinding('@.disabled')
	public disableAnimations = document.createElement('div').style['animation'] == null;
  
  .....
}
```

This will disable animations in problematic browsers.

**Comment by ghernandez81** [comment]
@janousek i fixed my issue with using NoopAnimationsModule in my unit-test

**Comment by pedep** [comment]
I think you can get pretty far by adding some vendor prefix checks at these 2 points
https://github.com/angular/angular/blob/6482f6f0fe16a488ec1509faca76015315ed8d29/packages/animations/browser/src/render/css_keyframes/css_keyframes_driver.ts#L43

https://github.com/angular/angular/blob/cf0968f98e844043a0f6c2548201f3c0dfd329a7/packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts#L9


maybe simply something like this?
```typescript
function vendorPrefix() {
  return ['', '-webkit-', '-moz-', '-o-'].find((prefix) =>
    document.body.style[prefix+'animation'] !== undefined || ''
  )
}

// [...]
let keyframeStr = `@${vendorPrefix()}keyframes ${name} {\n`; 
// [...]
const ANIMATION_PROP = vendorPrefix()+'animation'; 
```
 This approach requires calls to `setAnimationStyle` to use the lowercase-dashes version of the css properties
https://github.com/angular/angular/blob/cf0968f98e844043a0f6c2548201f3c0dfd329a7/packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts#L43
https://github.com/angular/angular/blob/cf0968f98e844043a0f6c2548201f3c0dfd329a7/packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts#L75

Thoughts?

**Comment by Brandinga** [comment]
have the same problem:
- TypeError: Cannot call method 'split' of undefined
- TypeError: Cannot call method 'trim' of undefined

User-Agent:
Mozilla/5.0 (Linux; Android 4.4.2; Doro Liberto 820 Build/KOT49H) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/30.0.0.0 Mobile Safari/537.36

FYI duplicate issue #25563

**Comment by Rostyy** [comment]
@omieliekh your solution will break angular material dropdowns in Safari.

**Comment by dottodot** [comment]
Is there still no fix for this?

**Comment by BojanKogoj** [comment]
There is a pull request opened (https://github.com/angular/angular/pull/29709), but nothing has been done about it.

@matsko any chance this will be merged?

**Comment by pascaliske** [comment]
This error also seems to occur in Chrome 70 on Windows 10 because I just got the same error message in my client's Angular app through Sentry: https://sentry.io/share/issue/8c7da85210424e63b96c8f7a10bc9435/

**Comment by vlaco** [comment]
Is there any hope for this being solved soon?

**Comment by danwhitston** [comment]
So, this has been blowing up our Sentry logging, and it's an absolutely tiny fix with a PR that has no chance of causing issues, and has been sitting in the queue forever. Seems unfortunate that it's taken over 2 years and several conversations to fail to change a single reference.

**Comment by petebacondarwin** [comment]
@danwhitston - sorry that this has not yet been solved. Now that @matsko has left the team, such issues are not getting as much attention as they deserve. Are you able to test the fix given in #29709 for your scenario? If it works for you, I would be willing to try to get it merged.

**Comment by AndrewKushnir** [comment]
Closing this issue since corresponding PR #29709 was merged and should be available in the next Angular release (v11.0.3). Thank you.
